smartsnippets/models.py

```diff
@@ -269,5 +271,9 @@ def remove_cached_variables(instance, **kwargs):
     if key in cache:
         cache.delete(key)
 
+def clean_variable_name(variable_name):
+    """ Convert the variable name to a name that can be used as a template variable. """
+    return re.sub(r"[^a-zA-Z0-9_]", '', variable_name.replace(' ', '_'))
```

Comment:
I also think that this would be the best solution, but it was not accepted. The requirements are that this should be done under the hood.

As for the special characters, they are not allowed in template variables at all. Existing variables with special characters and spaces do not work anyway. The only practical case that might be confusing for a user would be things like "my variable" transformed on save to "my_variable".